src/main/java/jenkins/plugins/logstash/LogstashPayloadProcessor.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2017 Red Hat inc, and individual contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.plugins.logstash;
+
+import net.sf.json.JSONObject;
+
+/**
+ * Interface describing processors of persisted payload.
+ */
+public interface LogstashPayloadProcessor {
+    /**
+     * Modifies a JSON payload compatible with the Logstash schema.
+     *
+     * @param payload the JSON payload that has been constructed so far.
+     * @return The formatted JSON object, can be null to ignore this payload.
+     */
+    JSONObject process(JSONObject payload) throws Exception;
```

Comment:
I think we actually only need the finish() method, not process().

Reply:
@jakub-bochenski 
Currently, `finish()` uses `process()`.  How would you like it to be instead?

Reply:
AFAIR finish only uses process to call it with null arg. Anyhow check the comment above (https://github.com/jenkinsci/logstash-plugin/pull/74#discussion_r269633076) it might make this point moot